shell/app-shell/elements/arc-host.js

```diff
@@ -50,7 +51,20 @@ class ArcHost extends Xen.Debug(Xen.Base, log) {
     // empty serialization which is ''
     if (id && context && pendingSerialization != null) {
       state.pendingSerialization = null;
+      // TODO(sjmiles): if we `_setState` we trigger invalidation if the return
```

Comment:
This looks like the answer to my question above, I just want to make sure we talk about doing this soon.